swap/src/protocol/alice/spot_price.rs

```diff
@@ -246,12 +281,15 @@ pub enum Error {
         balance: monero::Amount,
         buy: bitcoin::Amount,
     },
-
     #[error("Failed to fetch latest rate")]
     LatestRateFetchFailed(#[source] Box<dyn std::error::Error + Send + 'static>),
-
     #[error("Failed to calculate quote: {0}")]
     SellQuoteCalculationFailed(#[source] anyhow::Error),
+    #[error("Blockchain networks did not match, we are on {asb:?}, but request from {cli:?}")]
```

Comment:
OK to use Debug print for the error or will that break the log format?

Reply:
```
May 12 10:43:42.299 ERROR swap::protocol::alice::spot_price::tests: Blockchain networks did not match, we are on BlockchainNetwork { bitcoin: Testnet, monero: Stagenet }, but request from BlockchainNetwork { bitcoin: Mainnet, monero: Mainnet }

```

Reply:
How does it look if you enable json logging? Imho it should still work but it might escape the brackets.

Reply:
```
{
  "timestamp": "May 12 17:32:46.156",
  "level": "ERROR",
  "fields": {
    "message": "Blockchain networks did not match, we are on BlockchainNetwork { bitcoin: Testnet, monero: Stagenet }, but request from BlockchainNetwork { bitcoin: Mainnet, monero: Mainnet }"
  },
  "target": "swap::protocol::alice::spot_price::tests"
}
```

That's fine I guess, the brackets are just within the string, I think that is valid.

Reply:
@bonomat 

> Imho it should still work but it might escape the brackets.

Time to learn how [thiserror](https://docs.rs/thiserror/1.0.24/thiserror/) works :)